pandas/tests/frame/methods/test_compare.py

```diff
@@ -238,17 +238,65 @@ def test_invalid_input_result_names(result_names):
         df1.compare(df2, result_names=result_names)
 
 
-def test_compare_ea_and_np_dtype():
-    # GH#44014
-    df1 = pd.DataFrame({"a": [4.0, 4], "b": [1.0, 2]})
-    df2 = pd.DataFrame({"a": pd.Series([1, pd.NA], dtype="Int64"), "b": [1.0, 2]})
+@pytest.mark.parametrize(
+    "df1,df2,expected",
+    [
+        (
+            pd.DataFrame({"a": [4.0, 4], "b": [1.0, 2]}),
```

Comment:
I meant parametrising single values, not the whole object. This is really hard to read

Reply:
i thought this would be easier to understand. I've changed it now. please take a look.